### ISSUE TITLE
PyQUBO generates wrong QUBO model?

### ISSUE THREAD

**Issue by jnjt** [recruit-communications/pyqubo#107]
I used PyQUBO with D-Wave SDK, and I got weird result.

The source code is here.
```py
from pprint import pprint as prt
from pyqubo import Binary, Constraint
import dimod

x = Binary("x")
y = Binary("y")
z = Binary("z")

sub = (x * y) + (y * z)
c = Constraint((sub - 1) ** 2, label="c")

H = -(x + y + z) + c * 100
model = H.compile()

# solve
bqm = model.to_bqm()
sampler = dimod.ExactSolver()
sampleset = sampler.sample(bqm)

# result
dec = model.decode_sample(sampleset.first.sample, vartype="BINARY")
prt(dec.sample)
prt(dec.constraints())
prt(dec.energy)
```

The result is here.
```
{'0*1': 0, 'x': 1, 'y': 1, 'z': 1}
{'c': (False, 1.0)}
-98.0
```

The output says that the lowest energy is -98 with x = y = z = 1. But with x = y = z = 1, the "c" will be 1, and then the "H" will be 97. Am I misunderstanding something?

My environment is below.
```
$ pip3 freeze
Deprecated==1.2.13
dimod==0.10.7
dwave-neal==0.5.8
dwave-preprocessing==0.3.1.post0
numpy==1.19.5
pyparsing==2.4.7
pyqubo==1.0.13
six==1.16.0
wrapt==1.13.2
```

**Comment by kotarotanahashi** [comment]
@jnjt Thank you for reporting the issue.
It looks like you don't choose the minimum sample at `sampleset.first.sample`
To get the optimal solution, I recommend using the following code insted.

```
from pprint import pprint as prt
from pyqubo import Binary, Constraint
import dimod

x = Binary("x")
y = Binary("y")
z = Binary("z")

sub = (x * y) + (y * z)
c = Constraint((sub - 1) ** 2, label="c")

H = -(x + y + z) + c * 100
model = H.compile()

# solve
bqm = model.to_bqm()
sampler = dimod.ExactSolver()
sampleset = sampler.sample(bqm)

# result
decoded_samples = model.decode_sampleset(sampleset)
best_sample = min(decoded_samples, key=lambda x: x.energy)
print(best_sample.sample)
print("energy=", best_sample.energy)
print("constraint=", best_sample.constraints())
{'0*1': 0, 'x': 0, 'y': 1, 'z': 1}
{'c': (True, 0.0)}
-2.0
```

**Comment by jnjt** [comment]
@kotarotanahashi 
Thank you for your reply. But the answer doesn't work for me (with pyqubo 1.0.13).
```
{'0*1': 0, 'x': 1, 'y': 1, 'z': 1}
{'c': (False, 1.0)}
-98.0
```

By the way, with pyqubo 1.0.12, the result is fine.
```
{'0*1': 0, 'x': 0, 'y': 1, 'z': 1}
{'c': (True, 0.0)}
-2.0
```

I also check the QUBO matrix with this code.
```py
from pprint import pprint as prt
from pyqubo import Binary, Constraint
import pprint

x = Binary("x")
y = Binary("y")
z = Binary("z")

sub = (x * y) + (y * z)
c = Constraint((sub - 1) ** 2, label="c")

H = -(x + y + z) + c * 100
model = H.compile()

qubo, offset = model.to_qubo()
pprint.pprint(qubo)
```

With pyqubo 1.0.13.
```
{('0*1', '0*1'): 15.0,
 ('x', '0*1'): -10.0,
 ('x', 'x'): -1.0,
 ('x', 'y'): -100.0,
 ('y', '0*1'): 200.0,
 ('y', 'y'): -1.0,
 ('z', '0*1'): -10.0,
 ('z', 'x'): 5.0,
 ('z', 'y'): -100.0,
 ('z', 'z'): -1.0}
```

With pyqubo 1.0.12.
```
{('0*1', '0*1'): 14.0,
 ('x', '0*1'): -10.0,
 ('x', 'x'): 0.0,
 ('y', '0*1'): 200.0,
 ('y', 'y'): -1.0,
 ('z', '0*1'): -10.0,
 ('z', 'x'): 5.0,
 ('z', 'y'): -100.0,
 ('z', 'z'): -1.0}
```

So, I guess pyqubo 1.0.13 has problem.

**Comment by kotarotanahashi** [comment]
@jnjt 
Indeed, I confirmed the same problem with 1.0.13. The next version (1.1.x) that will be released soon is fixed problem, I ask you to wait until then. Please use 1.0.12 instead, until then. 
Thank you for your reporting!

**Comment by kotarotanahashi** [comment]
@jnjt 
Now we released a new version of PyQUBO. The problem should be fixed in this version.
https://pypi.org/project/pyqubo/

**Comment by jnjt** [comment]
@kotarotanahashi 
Thanks, it's fixed with 1.1.0.
